### PR TITLE
Add support for decoding tiff images with cmyk pixel data

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This PR adds support for decoding tiff images with cmyk pixel data.

Related issue: #2263

### CODE REVIEW

File: src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabTiffColor{TPixel}.cs [hunk 1]
```diff
@@ -31,8 +31,8 @@ public override void Decode(ReadOnlySpan<byte> data, Buffer2D<TPixel> pixels, in
             for (int x = 0; x < pixelRow.Length; x++)
             {
                 float l = (data[offset] & 0xFF) * 100f * Inv255;
-                var lab = new CieLab(l, (sbyte)data[offset + 1], (sbyte)data[offset + 2]);
-                var rgb = ColorSpaceConverter.ToRgb(lab);
+                CieLab lab = new(l, (sbyte)data[offset + 1], (sbyte)data[offset + 2]);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabTiffColor{TPixel}.cs:34]: I'm curious. Why are we casting to `sbyte` here? CieLab takes `float` for the constructor parameters. 

The `L` component ranges from `0-100`
Generally, the `A` and `B` components are unbounded though it's common to clamp them to `−128` to `127`

_Note. I just realized the property notes on the type are incorrect, saying `-100` to `100` for the `A` and `B` components._

https://en.wikipedia.org/wiki/CIELAB_color_space

Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabTiffColor{TPixel}.cs:34]: I'm curious to how the data is encoded here. 

The CieLab  `A`, `B` values usually range from −128 to 127 when working with integer values.  
Casting to `sbyte` and using overflow is a neat trick but you have to be careful.

**Review by JimBobSquarePants — APPROVED** [review]
LGTM 👍

File: src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabTiffColor{TPixel}.cs (hunk 1 above)
Comment (brianpopow) [reply src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabTiffColor{TPixel}.cs:34]: To cite the spec:

```
Tiff is defined such that each quantity be encoded with 8 bits. This provides 256 levels of L* lightness;
256 levels (+/- 127) of a*, and 256 levels (+/- 127) of b*.  
...
Limiting the theoretically unbounded a* and b*
ranges to +/- 127 allows encoding in 8 bits without eliminating any but the most
saturated self-luminous colors. It is anticipated that the rare specialized applications
requiring support of these extreme cases would be unlikely to use CIELAB
anyway. All object colors, in fact all colors within the theoretical MacAdam limits,
fall within the +/- 127 a*/b* range.
```

**Pull request merged**